src/main/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImpl.java

```diff
@@ -0,0 +1,1202 @@
+            LOG.error("Error persisting document # " + document.getDocumentHeader().getDocumentNumber() + " " + e.getMessage(), e);
+            throw new RuntimeException("Error persisting document # " + document.getDocumentHeader().getDocumentNumber() + " " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * @see AdvanceDepositService#routeAdvanceDepositDocuments()
+     */
+    public boolean routeAdvanceDepositDocuments() {
+        List<String> documentIdList;
+
+        try {
+            documentIdList = retrieveAdvanceDepositDocumentsToRoute(KewApiConstants.ROUTE_HEADER_SAVED_CD);
+        } catch (WorkflowException | RemoteException e) {
+            LOG.error("Error retrieving advance deposit documents for routing: " + e.getMessage(), e);
+            throw new RuntimeException(e.getMessage(), e);
+        }
+
+        if (LOG.isInfoEnabled()) {
+            LOG.info("Advance deposit to Route: " + documentIdList);
+        }
+
+        for (String documentId : documentIdList) {
+            try {
+                AdvanceDepositDocument advanceDocument = (AdvanceDepositDocument) documentService.getByDocumentHeaderId(documentId);
+                if (LOG.isInfoEnabled()) {
+                    LOG.info("Routing advance deposit document # " + documentId + ".");
+                }
+                documentService.prepareWorkflowDocument(advanceDocument);
+                workflowDocumentService.route(advanceDocument.getDocumentHeader().getWorkflowDocument(), "document routed by achIncome batch job", null);
+            } catch (WorkflowException e) {
+                LOG.error("Error routing document # " + documentId + " " + e.getMessage());
+                throw new RuntimeException(e.getMessage(), e);
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Returns a list of all initiated but not yet routed advance deposit documents, using the KualiWorkflowInfo service.
+     * @return a list of advance deposit documents to route
+     */
+    protected List<String> retrieveAdvanceDepositDocumentsToRoute(String statusCode) throws WorkflowException, RemoteException {
+        List<String> documentIds = new ArrayList<String>();
+
+        List<DocumentStatus> routeStatuses = new ArrayList<DocumentStatus>();
+        routeStatuses.add(DocumentStatus.fromCode(statusCode));
+
+        Person systemUser = getPersonService().getPersonByPrincipalName(KFSConstants.SYSTEM_USER);
+        String principalName = systemUser.getPrincipalName();
+
+        DocumentSearchCriteria.Builder criteria = DocumentSearchCriteria.Builder.create();
+        criteria.setDocumentTypeName(KFSConstants.FinancialDocumentTypeCodes.ADVANCE_DEPOSIT);
+        criteria.setDocumentStatuses(routeStatuses);
+        criteria.setInitiatorPrincipalName(principalName);
+
+        DocumentSearchResults results = KewApiServiceLocator.getWorkflowDocumentService().documentSearch(systemUser.getPrincipalId(), criteria.build());
```

Comment:
Not sure of any easy iterative means to get the desired documents with that criteria. Some of the 1099/1042S code does have an iterative solution to get around the doc search limit; however, it relies on knowing all the desired document IDs in advance, and may need tweaking in the future if we decide to change the limit (due to how it was implemented). If you're just interested in the document IDs, then perhaps you could perform a query or BO/CriteriaLookup service call involving the AD document table and the KFS doc header table.